xdsl/dialects/arm_neon.py

```diff
@@ -231,7 +242,9 @@ def assembly_line_args(self):
             VectorWithArrangement(self.d, self.arrangement),
             VectorWithArrangement(self.s1, self.arrangement),
             VectorWithArrangement(
-                self.s2, self.arrangement, index=self.scalar_idx.value.data
+                self.s2,
+                self.arrangement,
+                self.scalar_idx.value.data if self.scalar_idx is not None else None,
```

Comment:
```suggestion
                self.scalar_idx and self.scalar_idx.value.data
```
Possibly cooking too hard

Reply:
running into problems with this one (```Argument of type "IntegerAttr[IntegerType] | int | None" cannot be assigned to parameter "index" of type "int | None"```), not sure if I can get around it

Reply:
Ok, let's ignore this suggestion